tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/phases/TornadoHalfFloatReplacement.java

```diff
@@ -112,39 +155,121 @@ protected void run(StructuredGraph graph, TornadoHighTierContext context) {
         replaceMultHalfFloatNodes(graph);
         replaceDivHalfFloatNodes(graph);
 
+        // add after the loadindexedvector nodes the marker node to fix the offset of its read
+
+        for (LoadIndexedVectorNode loadIndexedVectorNode : graph.getNodes().filter(LoadIndexedVectorNode.class)) {
+            if (loadIndexedVectorNode.getOCLKind().isHalf()) {
+                VectorHalfRead vectorHalfRead;
+                if (loadIndexedVectorNode.index() instanceof ConstantNode) {
+                    ConstantNode offset = (ConstantNode) loadIndexedVectorNode.index();
+                    int offsetValue = Integer.valueOf(offset.getValue().toValueString());
+                    vectorHalfRead = graph.addWithoutUnique(new VectorHalfRead(offsetValue));
+                } else {
+                    vectorHalfRead = graph.addWithoutUnique(new VectorHalfRead());
+                }
+                graph.addAfterFixed(loadIndexedVectorNode, vectorHalfRead);
+            }
+        }
+
+        for (VectorValueNode vectorValueNode : graph.getNodes().filter(VectorValueNode.class)) {
+            if (vectorValueNode.getOCLKind().isHalf()) {
+                for (Node vectorElement : vectorValueNode.inputs()) {
+                    if (vectorElement instanceof VectorLoadElementNode) {
+                        VectorLoadElementNode vectorLoad = (VectorLoadElementNode) vectorElement;
+                        VectorLoadElementNode vectorLoadShort = new VectorLoadElementNode(OCLKind.SHORT, vectorLoad.getVector(), vectorLoad.getLaneId());
```

Comment:
Why is the type `SHORT` inserted here? Can we use a FLOAT instead? My understanding is that this is just to the internal representation in TornadoVM before the code gen.

Reply:
It's short so that the offset for reading is automatically calculated to be 2.

Reply:
Instead, I can use `OCLKind.HALF`, which is equivalent to what we have in the same phase for the SPIR-V backend. Internally, it's still the same though, because the `HALF` is still declared as short: 
`HALF(2, java.lang.Short.TYPE)`

Reply:
Changed it to `OCLKind.HALF`